Fix memory leak, add experimental setAll_P()

Koepel opened the pull request:
Merging resolves https://github.com/Simsso/ShiftRegister74HC595/issues/14

Koepel committed:
Update ShiftRegister74HC595.h

Koepel committed:
Update ShiftRegister74HC595.cpp

Simsso commented:
Thank you, I really appreciate you took the time and fixed the code. I'll review and test during the next days and merge afterwards. Once again, thank you. Such high quality contributions are rare! 🥇

Simsso commented on src/ShiftRegister74HC595.cpp:
```diff
@@ -14,46 +14,58 @@ ShiftRegister74HC595::ShiftRegister74HC595(int numberOfShiftRegisters, int seria
+    memcpy_P( _digitalValues, p, _numberOfShiftRegisters);
+    updateRegisters();
+}
+#endif
```
I have tried to compile with the Arduino IDE version 1.8.9 on Mac, for Arduino Uno and am getting the following error:

```
src/ShiftRegister74HC595.cpp: In function 'void setAll_P(const uint8_t*)':
src/ShiftRegister74HC595.cpp:63:15: error: '_digitalValues' was not declared in this scope
     memcpy_P( _digitalValues, p, _numberOfShiftRegisters);
               ^
src/ShiftRegister74HC595.cpp:63:34: error: '_numberOfShiftRegisters' was not declared in this scope
     memcpy_P( _digitalValues, p, _numberOfShiftRegisters);
                                  ^
src/ShiftRegister74HC595.cpp:64:21: error: 'updateRegisters' was not declared in this scope
     updateRegisters();
                     ^
src/ShiftRegister74HC595.cpp: In member function 'void ShiftRegister74HC595::updateRegisters()':
src/ShiftRegister74HC595.cpp:90:55: error: 'digitalValues' was not declared in this scope
         shiftOut(_serialDataPin, _clockPin, MSBFIRST, digitalValues[i]);
                                                       ^
exit status 1
Error compiling for board Arduino/Genuino Uno.
```

Simsso commented on src/ShiftRegister74HC595.cpp:
```diff
@@ -71,9 +83,15 @@ void ShiftRegister74HC595::set(int pin, uint8_t value)
 {
-	 setAll(_digitalValues);
+    for (int i = _numberOfShiftRegisters - 1; i >= 0; i--) {
+        shiftOut(_serialDataPin, _clockPin, MSBFIRST, digitalValues[i]);
```
Little typo, should be `_digitalValues`, here.

Simsso replied:
I have taken care of that one in this commit: https://github.com/Simsso/ShiftRegister74HC595/pull/15/commits/c496675fee4e48a53c5de7297f8fd9d09e0ca41b

Simsso committed:
Fix typo '_digitalValues'

Koepel commented:
O, I'm sorry. Updating the shift registers with the data from the heap is with `_digitalValues` of course.
Is there something I should do ?

Simsso commented:
Yes, concerning [this error](https://github.com/Simsso/ShiftRegister74HC595/pull/15#discussion_r268982245), do you know what the cause is? Where did you get it to compile? If not I'd suggest removing the experimental parts for now and would merge the PR with the other improvements only.

Koepel committed:
getAll_P -> ShiftRegister74HC595::getAll_P

Koepel commented:
I really messed up big time. The function getAll_P did not know the variables of the class, because it was not part of the class. Changed `getAll_P(` to `ShiftRegister74HC595::getAll_P(`.
This time I removed my test versions and carefully checked that my fork is okay.

The Arduino Zero and Arduino MKR allow to use the keyword `PROGMEM` and also the function `memcpy_P()`. That makes the source compatible with the Arduino Uno. Even the ESP8266 libraries allow to use `PROGMEM`. Perhaps the `#ifdef __AVR__` and `#endif` can be removed, but I don't know if it is guaranteed accepted for all Arduino boards. It is easier when the user copies the PROGMEM data into a local buffer and just use `setAll()`.

Simsso committed:
Increment library version

Simsso commented:
Thank you once again. I've done some testing with actual hardware and the changes look very good. Merged ✅

Koepel commented:
Sorry once again. I've should have tested the final pull request instead of a test version on my computer.

You could add a destructor with a `free(this->data)` to your `ShiftRegister-PWM-Library` as well.

Simsso commented:
I think we're freeing the allocated memory here, don't we: https://github.com/Simsso/ShiftRegister74HC595/blob/master/src/ShiftRegister74HC595.cpp#L44

Koepel commented:
Yes, I meant for your "PWM" library.
